detection_rules/rule_loader.py

```diff
@@ -239,6 +239,13 @@ def add_deprecated_rule(self, rule: DeprecatedRule):
         self.deprecated.name_map[rule.name] = rule
         self.deprecated.rules.append(rule)
 
+    def get_bbr(self) -> List[TOMLRule]:
+        bbr_list = []
+        for rule in self.rules:
+            if "building_block_type" in rule.contents.data.to_dict().keys():
+                bbr_list.append(rule)
+        return bbr_list
+
```

Comment:
You shouldn't need this.

Reply:
Unfortunately it is not as simple as detection a rule of type `building_block` the building block rules need to retain their rule type e.g. `QueryRuleData` etc. Filters are also will not function currently as the information is not in the metadata but in the rule contents which are different across rule types.

Reply:
do you want to try generalizing the `metadata_filter` to be more generic?

`path_getter` allows dotted conversion, which is useful since rule data is heavily nested

```diff
-def metadata_filter(**metadata) -> Callable[[TOMLRule], bool]:
+def metadata_filter(**filter_data, target: Literal['metadata', 'data', 'transform' = 'metadata') -> Callable[[TOMLRule], bool]:

    """Get a filter callback based off rule metadata"""
+    flt = dict_filter(filter_data)

    def callback(rule: TOMLRule) -> bool:
+      target = getattr(rule.contents, target, None)
+      if not target:
+          return False
        target_dict = target.to_dict()
        return flt(target_dict)

    return callback
```

(I haven't tested it)

Reply:
Also, do we think the existing `production_filter` should exclude BBR? We should search every where it is used